Sources/ConverterScreen.swift

```diff
@@ -798,3 +772,39 @@ class ConverterScreen: UIViewController {
     }
 }
 
+extension ConverterScreen: DiagramResultDelegate {
+    func currenciesFromDiagramToConverter(curInput: String?, curOutput1: String?, curOutput2: String?, curOutput3: String?) {
+        if let curInputInst = curInput {
+            self.chosenCurShortName = curInput
+        }
+        if let curOutputInst1 = curOutput1 {
+            self.chosenCurShortName1 = curOutput1
+            self.outputCurrencyLabel1.text = updateOutputCurrencyLabel(chosenCurrency: self.chosenCurShortName1)
+        }
+        if let curOutputInst2 = curOutput2 {
+            self.chosenCurShortName2 = curOutput2
+            self.outputCurrencyLabel2.text = updateOutputCurrencyLabel(chosenCurrency: self.chosenCurShortName2)
+            self.counterOfClick = 0
+            self.addCurrency()
+        }
+        if let curOutputInst3 = curOutput3 {
+            self.chosenCurShortName3 = curOutput3
+            self.outputCurrencyLabel3.text = updateOutputCurrencyLabel(chosenCurrency: self.chosenCurShortName3)
+            for counter in 0...1 {
```

Comment:
Why `0...1`?

Reply:
Because 0 and 1 moves constraints of buttonDiagramPage and buttonRateHistory lower in the method updateOutputCurrencyLabel of ConverterScreen class.